app/controllers/general.php

```diff
@@ -51,7 +51,11 @@
     $port = \parse_url($request->getOrigin($referrer), PHP_URL_PORT);
 
     $refDomain = (!empty($protocol) ? $protocol : $request->getProtocol()).'://'.((\in_array($origin, $clients))
-        ? $origin : 'localhost') . (!empty($port) ? ':'.$port : '');
+        ? $origin : 'localhost').(!empty($port) ? ':'.$port : '');
+    
+    $refDomain = (!$route->getLabel('origin', false))  // This route is publicly accessible
```

Comment:
Added this feature to be able to have specific routes allow any cross origin communication. This is mainly used in the homepage itself for reporting.